component/discovery/kubernetes_crds/promcfg.go

```diff
@@ -0,0 +1,388 @@
+	// 		if s, ok := store.BasicAuthAssets["apiserver"]; ok {
+	// 			k8sSDConfig = append(k8sSDConfig, yaml.MapItem{
+	// 				Key: "basic_auth", Value: yaml.MapSlice{
+	// 					{Key: "username", Value: s.Username},
+	// 					{Key: "password", Value: s.Password},
+	// 				},
+	// 			})
+	// 		}
+	// 	}
+
+	// 	if apiserverConfig.BearerToken != "" {
+	// 		k8sSDConfig = append(k8sSDConfig, yaml.MapItem{Key: "bearer_token", Value: apiserverConfig.BearerToken})
+	// 	}
+
+	// 	if apiserverConfig.BearerTokenFile != "" {
+	// 		k8sSDConfig = append(k8sSDConfig, yaml.MapItem{Key: "bearer_token_file", Value: apiserverConfig.BearerTokenFile})
+	// 	}
+
+	// 	k8sSDConfig = cg.addAuthorizationToYaml(k8sSDConfig, "apiserver/auth", store, apiserverConfig.Authorization)
+
+	// 	// TODO: If we want to support secret refs for k8s service discovery tls
+	// 	// config as well, make sure to path the right namespace here.
+	// 	k8sSDConfig = addTLStoYaml(k8sSDConfig, "", apiserverConfig.TLSConfig)
+	// }
+
+	// TODO:
+	if attachMetadata != nil {
+
+		//k8sSDConfig = cg.WithMinimumVersion("2.35.0").AppendMapItem(k8sSDConfig, "attach_metadata", yaml.MapSlice{
+		//	{Key: "node", Value: attachMetadata.Node},
+		//})
+	}
+	return cfg
+}
+
+func (cg *configGenerator) generatePodMonitorConfig(m *v1.PodMonitor, ep v1.PodMetricsEndpoint, i int) *config.ScrapeConfig {
+	c := config.DefaultScrapeConfig
+	cfg := &c
+	cfg.ScrapeInterval = config.DefaultGlobalConfig.ScrapeInterval
+	cfg.ScrapeTimeout = config.DefaultGlobalConfig.ScrapeTimeout
+	cfg.JobName = fmt.Sprintf("podMonitor/%s/%s/%d", m.Namespace, m.Name, i)
+	cg.addHonorLabels(cfg, ep.HonorLabels)
+	cg.addHonorTimestamps(cfg, ep.HonorTimestamps)
+
+	cfg.ServiceDiscoveryConfigs = append(cfg.ServiceDiscoveryConfigs, cg.generateK8SSDConfig(m.Spec.NamespaceSelector, m.Namespace, promk8s.RolePod, m.Spec.AttachMetadata))
+
+	if ep.Interval != "" {
+		// TODO: correct way to convert the durations?
+		cfg.ScrapeInterval, _ = model.ParseDuration(string(ep.Interval))
+	}
+	if ep.ScrapeTimeout != "" {
+		cfg.ScrapeInterval, _ = model.ParseDuration(string(ep.ScrapeTimeout))
+	}
+	if ep.Path != "" {
+		cfg.MetricsPath = ep.Path
+	}
+	if ep.ProxyURL != nil {
+		u, _ := url.Parse(*ep.ProxyURL)
+		cfg.HTTPClientConfig.ProxyURL = commonConfig.URL{URL: u}
+	}
+	if ep.Params != nil {
+		cfg.Params = ep.Params
+	}
+	if ep.Scheme != "" {
+		cfg.Scheme = ep.Scheme
+	}
+	if ep.FollowRedirects != nil {
+		cfg.HTTPClientConfig.FollowRedirects = *ep.FollowRedirects
+	}
+	if ep.EnableHttp2 != nil {
+		cfg.HTTPClientConfig.EnableHTTP2 = *ep.EnableHttp2
+	}
+	//if ep.TLSConfig != nil {
+	//TODO:
+	//cfg = addSafeTLStoYaml(cfg, m.Namespace, ep.TLSConfig.SafeTLSConfig)
+	//}
+
+	//TODO: Secret store needs to be figured out
+	// if ep.BearerTokenSecret.Name != "" {
+	// 	if s, ok := store.TokenAssets[fmt.Sprintf("podMonitor/%s/%s/%d", m.Namespace, m.Name, i)]; ok {
+	// 		cfg = append(cfg, yaml.MapItem{Key: "bearer_token", Value: s})
+	// 	}
+	// }
```

Comment:
Can't the component request the content of secrets from the K8s API client? It would search in the same namespace as the PodMonitor.